docs/source/developing-volttron/developing-agents/agent-development.rst

```diff
@@ -629,6 +629,108 @@ The VOLTTRON team has come up with a number of methods to help users develop mor
    safely.
 
 
+Building a resilient API
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Many agents expose an API (application programming interface) which can be used by other agents on the platform.   To
+ensure an agent's API is robust, the user should take care to validate input and secure the endpoints.
+
+
+Type-hints
+""""""""""
+
+`Type-hints <https://docs.python.org/3/library/typing.html>`_ can be used in function definitions to help the user
+determine what the agent expects as input.
+
+.. warning::
+
+ Type-hints do not validate the type of data input to a function call, they are merely suggestions about what the
+ function is expecting.
+
+* To specify the type expected as input:
+
+.. code-block:: python
+
+   # When calling this RPC method, the user should supply a string as input
+   @RPC.export
+   def type_hint_rpc(input_string: str):
+
+* To specify the type of function output:
+
+.. code-block:: python
+
+   # This demonstrates a function that expects a string as input and that will return an integer value
+   @RPC.export
+   def type_hint_rpc(input_string: str) -> int:
+
+* Specifying multiple types:
+
+.. code-block:: python
+
+    # Here our function expects either a string or dictionary representation of some JSON data
+    @RPC.export
+    def type_hint_rpc(input_json: Union[str, dict]) -> str:
+
+* To specify an optional argument with None as the default value:
+
+.. code-block:: python
+
+    # 'Optional' is used here to specify that either a string should be passed or the default value 'None' will be
+    # used
+    # 'Optional' can be used in combination with 'Union' for optional arguments which expect one of multiple types
+    @RPC.export
+    def type_hint_rpc(optional_input: Optional[str] = None) -> str:
```

Comment:
This example doesn't show the Union with Optional argument like the last comment suggests.